src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,122 @@
+            statement.setString(FIRST_INDEX, book.getTitle());
+            statement.setBigDecimal(SECOND_INDEX, book.getPrice());
+            if (statement.executeUpdate() < AT_LEAST_ONE_ROW) {
+                throw new RuntimeException("Excepted "
+                        + "to insert at least one row, but inserted 0 rows.");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(FIRST_INDEX, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t add new book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_BY_ID_REQUEST)) {
+            statement.setLong(FIRST_INDEX, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Book book = new Book();
+                book.setId(id);
+                book.setTitle(resultSet.getString(TITLE));
+                book.setPrice(resultSet.getObject(PRICE, BigDecimal.class));
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t find book by id: " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_ALL_REQUEST)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Book book = new Book();
+                book.setId(resultSet.getObject(ID, Long.class));
+                book.setTitle(resultSet.getString(TITLE));
+                book.setPrice(resultSet.getObject(PRICE, BigDecimal.class));
+                books.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't receive books.", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(UPDATE_REQUEST)) {
+            statement.setString(FIRST_INDEX, book.getTitle());
+            statement.setBigDecimal(SECOND_INDEX, book.getPrice());
+            statement.setLong(THIRD_INDEX, book.getId());
+            int affectedRow = statement.executeUpdate();
+            if (affectedRow < AT_LEAST_ONE_ROW) {
+                throw new RuntimeException("Expected at least one row, but inserted 0 rows.");

```

Comment:
Probably we can make our error messages more generic and extract them to the constants as well